### ISSUE TITLE
Release v1.4.0 changes

### ISSUE THREAD

**Issue by ashleve** [ashleve/lightning-hydra-template#308]
Changes considered for release v1.4.0

Feel free to share your thoughts.

Available at:
https://github.com/ashleve/lightning-hydra-template/tree/release/1.4

PR: https://github.com/ashleve/lightning-hydra-template/pull/322

- [x] adapt template to hydra 1.2 - no more changing the working directory by default
- [x] rename `test.py` and `test.yaml` to `eval.py` and `eval.yaml`, so as to avoid confusion with project tests
- [x] move `train.py` and `eval.py` inside `src/`
- [x] add `pyrootutils` package for setting up the project root in `train.py` and `eval.py`
- [x] rename `pipelines` to `tasks`
- [x] separate folder for tasks
- [x] add `task_name` to config which determines hydra output folder path
- [x] introduce `@task_wrapper` decorator for applying utilities before and after the task is executed
- [x] rename `config` to `cfg` since it's the standard naming convention in hydra
- [x] introduce `utils.instantiate_callbacks()` and `utils.instantiate_loggers()` to reduce the boilerplate
- [x] introduce `utils.get_metric_value()` for retrieving optimized metric.
- [x] add `utils.close_loggers()` for preventing multirun failure when using neptune, comet or mlflow 
- [x] add `pyupgrade` to pre-commit hooks (automatically upgrading python syntax to newer version)
- [x] add `bandit` to pre-commit hooks (security linter)
- [x] add tests covering ddp, multirun, loggers, resuming training and evaluation
- [x] implement CI tests with github actions
- [x] add code coverage measuring to CI
- [x] add Dependabot
- [x] add trainer config for mps accelerator (Accelerated PyTorch Training on Mac)
- [x] add `hydra.mode=MULTIRUN` to hparams_search configs
- [x] update `mnist_opuna.yaml` for compatibility with new search space syntax in hydra 1.2
- [x] replace `setup.cfg` with `pyproject.toml` since it's a more versatile standard (PEP 518)
- [x] disable callbacks in debug configs by default (fixes `debug/overfit.yaml`)
- [x] split utils into several files: `utils.py`, `rich_utils.py`, `pylogger.py`
- [x] remove deprecated `trainer.gpus` argument, add `trainer.accelerator` and `trainer.devices`
- [x] add separate trainer configs for GPU and simulating DDP on CPU
- [x] disable hydra command line debug logger in debug configs by default
- [x] add `deterministic` flag to trainer config 
- [x] add enforcing tags with rich prompting to utils
- [x] remove experiment name since tags and task name are enough
- [x] move extra config utils to `configs/extras/default.yaml`
- [x] split callbacks config into multiple files
- [x] add setup.py 
- [x] add codespell linter 
- [x] update readme

**Comment by johnnynunez** [comment]
Multi run is now native hydra 1.2
https://hydra.cc/docs/tutorials/basic/running_your_app/multi-run/

**Comment by johnnynunez** [comment]
@ashleve i'm trying to do work m1

**Comment by johnnynunez** [comment]
@ashleve 
<img width="942" alt="Screenshot 2022-05-26 at 18 22 34" src="https://user-images.githubusercontent.com/22727137/170530820-97f7b047-999e-4119-b6ee-ea40fed7a181.png">

change gpus: 1 that is deprecated by : 

accelerator: 'mps'
devices: 1

**Comment by johnnynunez** [comment]
@ashleve I have this warning 
```
wandb: WARNING Path {paths.output_dir}/wandb/ wasn't writable, using system temp directory.
wandb: WARNING Path {paths.output_dir}/wandb/ wasn't writable, using system temp directory
```

**Comment by ashleve** [comment]
@johnnynunez Thanks, I fixed the paths.

`train.log` will be fixed when this PR is merged https://github.com/facebookresearch/hydra/pull/2242

**Comment by johnnynunez** [comment]
@ashleve DDP Fixed in hydra v1.3 https://github.com/facebookresearch/hydra/issues/2070
